lib/spark.js

```diff
@@ -58,8 +64,30 @@ export default class Spark {
             stats.firstByteAt = new Date()
           }
           stats.byteLength += value.byteLength
+
+          // We want to limit how large content we are willing to download.
+          // 1. To make sure we don't spend too much time (and network bandwidth) on a single task,
+          //    so that we can complete more tasks per round
+          // 2. Until we have streaming hashes, we need to keep the entire payload in memory, and so
+          //    we need to put an upper limit on how much memory we consume.
+          if (stats.byteLength > MAX_CAR_SIZE) {
+            stats.carTooLarge = true
+            break
+          }
+
+          const offset = carBuffer.byteLength
+          carBuffer.resize(offset + value.byteLength)
+          carBytes.set(value, offset)
+
           resetTimeout()
         }
+
+        if (!stats.carTooLarge) {
+          const digest = await crypto.subtle.digest('sha-256', carBytes)
```

Comment:
I decided to use SHA-256 instead of SHA-512 which I proposed in the past. 
- The digest is shorter and has the same length as Blake3 we want to switch to later.
- We don't really care about SHA-256's length extension attacks, because we will be switching to Blake3 anyways.